### PR TITLE
Add samples for each CRD

### PR DESCRIPTION
**Pull request by ryansteakley**
Added samples for each CRD for the sagemaker.services.k8s.aws controller.
Building off of https://github.com/aws-controllers-k8s/sagemaker-controller/pull/32
Rebased and squashed commits

### CODE REVIEW

**Comment by ryansteakley** [comment]
@RedbackThomson opened a new pr with squashed commits but did not forget about your [comments](https://github.com/aws-controllers-k8s/sagemaker-controller/pull/32#pullrequestreview-662388390) worry :)

File: samples/monitoring_schedule/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Bring Your Own Monitoring Schedule Sample
+
+This sample demonstrates how to start Monitoring scheduling using your own Monitor scheduling script, packaged in a SageMaker-compatible container, using the Amazon AWS Controllers for Kubernetes (ACK) service controller for Amazon SageMaker.   
+
+
+## Prerequisites
+
+This sample assumes that you have already configured an EKS cluster with the ACK operator. It also assumes that you have installed `kubectl` - you can find a link on our [installation page](To do). It also assumes you have a job of one of these types already created `DataQuality|ModelQuality|ModelBias|ModelExplainability`.
```
Comment (surajkota) [inline samples/monitoring_schedule/README.md:8]: link to the directories for job definitions and change the wording to say job definitions

File: samples/monitoring_schedule/my-monitoring-schedule.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+apiVersion: sagemaker.services.k8s.aws/v1alpha1
+kind: MonitoringSchedule
+metadata:
+  name: my-monitoring-schedule
+spec:
+  monitoringScheduleName: my-monitoring-schedule
```
Comment (surajkota) [inline samples/monitoring_schedule/my-monitoring-schedule.yaml:6]: make this a variable

File: samples/processing/my-processing-job.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+metadata:
+  name: <YOUR JOB NAME>
+spec:
+  processingJobName: <YOUR JOB NAME> 
+  # Name that will appear in SageMaker console
+  roleARN: <YOUR SAGEMAKER ROLE ARN> 
+  # A role with SageMaker and S3 access
+  # example arn:aws:iam::1234567890:role/service-role/AmazonSageMaker-ExecutionRole
+  appSpecification:
+    imageURI: 763104351884.dkr.ecr.us-west-2.amazonaws.com/pytorch-training:1.5.0-cpu-py36-ubuntu16.04
```
Comment (surajkota) [inline samples/processing/my-processing-job.yaml:12]: this is a DLC image not sagemaker 1-p
https://github.com/aws/deep-learning-containers/blob/master/available_images.md

File: samples/training/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# Bring Your Own Training Job Sample
+
+This sample demonstrates how to start training jobs using your own training script, packaged in a SageMaker-compatible container, using the Amazon AWS Controllers for Kubernetes (ACK) service controller for Amazon SageMaker.                     
+
+## Prerequisites
+
+This sample assumes that you have already configured an EKS cluster with the ACK operator. It also assumes that you have installed `kubectl` - you can find a link on our [installation page](To do).
+
+In order to follow this script, you must first create a training script packaged in a Dockerfile that is [compatible with Amazon SageMaker](https://docs.aws.amazon.com/sagemaker/latest/dg/amazon-sagemaker-containers.html). The [Distributed Mask R-CNN sample](https://github.com/awslabs/amazon-sagemaker-examples/tree/master/advanced_functionality/distributed_tensorflow_mask_rcnn), published by the SageMaker team, contains a predefined training script and helper bash scripts for reference.
```
Comment (surajkota) [inline samples/training/README.md:9]: mask rc-nn ? i dont think the sample is related to tensorflow or distributed training

File: samples/training/s3_sample_data_creator.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import shap
```
Comment (surajkota) [inline samples/training/s3_sample_data_creator.py:1]: afaik, this is not the right script for creating data, examples use mnist dataset

**Review by surajkota — CHANGES_REQUESTED** [review]

File: samples/job_definitions/model_quality/README.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
+# Model Quality Job Definition Sample
+
+This sample demonstrates how to start model-quality job definitions using your own model-quality script, packaged in a SageMaker-compatible container, using the Amazon AWS Controllers for Kubernetes (ACK) service controller for Amazon SageMaker.                     
+
+## Prerequisites
+
+This sample assumes that you have already configured an EKS cluster with the ACK operator. It also assumes that you have installed `kubectl` - you can find a link on our [installation page](To do).
+
+You will also need an [Endpoint](/samples/endpoint/README.md) configured in SageMaker and will also need to run baselining jobs using the Python SageMaker SDK before you can use it with ACK.
```
Comment (surajkota) [inline samples/job_definitions/model_quality/README.md:9]: ~~will also need to run baselining jobs using the Python SageMaker SDK before you can use it with ACK~~

you will need to run a baselining job to generate baseline statistics and constraints.

File: samples/job_definitions/model_explainability/README.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
+# Model Explainability Job Definition Sample
+
+This sample demonstrates how to start model-explainability job definitions using your own model-explainability script, packaged in a SageMaker-compatible container, using the Amazon AWS Controllers for Kubernetes (ACK) service controller for Amazon SageMaker.                     
+
+## Prerequisites
+
+This sample assumes that you have already configured an EKS cluster with the ACK operator. It also assumes that you have installed `kubectl` - you can find a link on our [installation page](To do).
+
+You will also need an [Endpoint](/samples/endpoint/README.md) configured in SageMaker and will also need to run baselining jobs using the Python SageMaker SDK before you can use it with ACK.
```
Comment (surajkota) [inline samples/job_definitions/model_explainability/README.md:9]: ~~will also need to run baselining jobs using the Python SageMaker SDK before you can use it with ACK~~

you will need to run a baselining job to generate baseline constraints.

File: samples/job_definitions/model_bias/README.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
+# Model Bias Job Definition Sample
+
+This sample demonstrates how to start model-bias job definitions using your own model-bias job definition script, packaged in a SageMaker-compatible container, using the Amazon AWS Controllers for Kubernetes (ACK) service controller for Amazon SageMaker.                     
+
+## Prerequisites
+
+This sample assumes that you have already configured an EKS cluster with the ACK operator. It also assumes that you have installed `kubectl` - you can find a link on our [installation page](To do).
+
+You will also need an [Endpoint](/samples/endpoint/README.md) configured in SageMaker.
```
Comment (surajkota) [inline samples/job_definitions/model_bias/README.md:9]: ~~will also need to run baselining jobs using the Python SageMaker SDK before you can use it with ACK~~

you will need to run a baselining job to generate baseline constraints.

File: samples/job_definitions/data_quality/README.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
+# Data Quality Job Definition Sample
+
+This sample demonstrates how to start data quality job definitions using your own data-quality-job-definitions script, packaged in a SageMaker-compatible container, using the Amazon AWS Controllers for Kubernetes (ACK) service controller for Amazon SageMaker.                     
+
+## Prerequisites
+
+This sample assumes that you have already configured an EKS cluster with the ACK operator. It also assumes that you have installed `kubectl` - you can find a link on our [installation page](To do).
+
+You will need an [Endpoint](/samples/endpoint/README.md) configured in SageMaker and also will need to run baselining jobs using the Python SageMaker SDK before you can use it with ACK.
```
Comment (surajkota) [inline samples/job_definitions/data_quality/README.md:9]: ~~will also need to run baselining jobs using the Python SageMaker SDK before you can use it with ACK~~

you will need to run a baselining job to generate baseline constraints.

it is possible to do it using ACK, later we can add samples for that

File: samples/processing/kmeans_preprocessing.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import pickle
```
Comment (surajkota) [inline samples/processing/kmeans_preprocessing.py:1]: add to pre-reqs - upload this script to an S3 bucket and say the path needs to be updated in 
```
# The input path of our source code
s3URI: s3://<YOUR BUCKET>/path
```

File: samples/model/README.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+## Create your Model
+To submit your prepared model specification, apply the specification to your EKS cluster as such:
+```
+$ kubectl apply -f my-model.yaml
+models.sagemaker.services.k8s.aws.amazon.com/my-model created
+```
+
+## List your Model
+To monitor the model status, you can use the following command:
```
Comment (surajkota) [inline samples/model/README.md:34]: Can we reword this text to change the word `monitor`? and list resources rather than a single resource

 List your Model`s`
Use the following command to list all [models/jobs/endpoints etc.] created using the ACK controller:


kubectl get models.sagemaker.services.k8s.aws ~~my-model~~


because list means multiple in general. for singular user, can run describe instead. Same for other resources

File: samples/model/README.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+### Updating the Model Specification
+
+In the `my-model.yaml` file, modify the placeholder values with those associated with your account and model.
+
+# Submitting your Model
+
+## Create your Model
+To submit your prepared model specification, apply the specification to your EKS cluster as such:
```
Comment (surajkota) [inline samples/model/README.md:27]: EKS -> Kubernetes
everywhere

File: samples/model/README.md [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+## Describe your Model
+To monitor the model in-depth once it has started, you can see the full status and any additional errors with the following command:
+```
+$ kubectl describe models.sagemaker.services.k8s.aws my-model
+```
+
+## Delete your Model
+To delete the model once it has started if errors occurred or for any reason with the following command:
```
Comment (surajkota) [inline samples/model/README.md:46]: To delete the model ~~once it has started~~

To delete the model, use the following command:

File: samples/model/README.md [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+A container image URL and tag looks has the following structure:
+```
+<account number>.dkr.ecr.<region>.amazonaws.com/<image name>:<tag>
+```
+
+### Updating the Model Specification
+
+In the `my-model.yaml` file, modify the placeholder values with those associated with your account and model.
+
+# Submitting your Model
```
Comment (surajkota) [inline samples/model/README.md:24]: nit: only the first line heading of the page should be H1, please make others H2 or smaller

File: samples/model/README.md [hunk 15]
```diff
@@ -0,0 +1,49 @@
+# Model Sample
+
+This sample demonstrates how to start models/create a model using your own model script, packaged in a SageMaker-compatible container, using the Amazon AWS Controllers for Kubernetes (ACK) service controller for Amazon SageMaker.                     
+
+## Prerequisites
+
+This sample assumes that you have already configured an EKS cluster with the ACK operator. It also assumes that you have installed `kubectl` - you can find a link on our [installation page](To do).
+
+## Preparing the Model Script
```
Comment (surajkota) [inline samples/model/README.md:9]: let's remove this line, it is confusing. content of this section fit fine under prereqs

File: samples/model/README.md [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+## List your Model
+To monitor the model status, you can use the following command:
+```
+$ kubectl get models.sagemaker.services.k8s.aws my-model
+```
+
+## Describe your Model
+To monitor the model in-depth once it has started, you can see the full status and any additional errors with the following command:
```
Comment (surajkota) [inline samples/model/README.md:40]: can we reword this line a little bit to change the words `monitor` and `in-depth`?

one possible line - 
To get more details about the resource once it's submitted, like checking the status, errors or parameters of the resource . . .

File: samples/model/my-model.yaml [hunk 17]
```diff
@@ -0,0 +1,18 @@
+apiVersion: sagemaker.services.k8s.aws/v1alpha1
+kind: Model
+metadata:
+  name: <YOUR MODEL NAME>
+spec:
+  modelName: <YOUR MODEL NAME>
+  primaryContainer:
+    containerHostname: xgboost
+    # The source of the model data
+    modelDataURL: s3://<YOUR BUCKET>/path
```
Comment (surajkota) [inline samples/model/my-model.yaml:10]: should we wrap S3 paths like this `s3://<YOUR BUCKET/PATH>` ?

**Review by surajkota — CHANGES_REQUESTED** [review]
minor comments, mostly around README

File: samples/model/my-model.yaml (hunk 17 above)
Comment (ryansteakley) [reply samples/model/my-model.yaml:10]: Helps to imply you need to change path as well.  Sounds good

File: samples/job_definitions/model_quality/README.md [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+
+To get more details about the Model-Quality Job Definition once it's submitted, like checking the status, errors or parameters of the Model-Quality Job Definition use the following command:
+```
+$ kubectl describe modelQualityjobdefinitions my-model-quality-job-definition
+```
+You can also check Status.ackResourceMetadata.Arn to verify the data quality job definition was created successfully.
+
+### Delete a Model-Quality Job Definition
+
+To delete the model-quality job definition once it has started if errors occurred or for any reason with the following command:
```
Comment (surajkota) [inline samples/job_definitions/model_quality/README.md:51]: reword

File: samples/job_definitions/model_quality/README.md (hunk 6 above)
Comment (surajkota) [reply samples/job_definitions/model_quality/README.md]: Apologies, statistics is only valid for data quality job definition

all other job definitions should say `you will need to run a baselining job to generate baseline constraints only`

**Review by surajkota — APPROVED** [review]
First PR - customer obsession ✔ ✔

Awesome work on the samples!!

File: samples/README.md [hunk 19]
```diff
@@ -0,0 +1,13 @@
+# Job Sample Overview
+
+This sample demonstrates how to start jobs using your own script, packaged in a SageMaker-compatible container, using the Amazon AWS Controllers for Kubernetes (ACK) service controller for Amazon SageMaker.
```
Comment (mbaijal) [inline samples/README.md:3]: nit: Should this be `these samples` ?

File: samples/README.md [hunk 20]
```diff
@@ -0,0 +1,13 @@
+# Job Sample Overview
+
+This sample demonstrates how to start jobs using your own script, packaged in a SageMaker-compatible container, using the Amazon AWS Controllers for Kubernetes (ACK) service controller for Amazon SageMaker.                     
+
+## Prerequisites    
+
+This sample assumes that you have already configured an Kubernetes cluster with the ACK operator. It also assumes that you have installed `kubectl` - you can find a link on our [installation page](To do).
```
Comment (mbaijal) [inline samples/README.md:7]: Can we be more specific - installed CRDs and controller instead of saying just operator here.

File: samples/endpoint/endpoint_config.yaml [hunk 21]
```diff
@@ -0,0 +1,26 @@
+apiVersion: sagemaker.services.k8s.aws/v1alpha1
```
Comment (mbaijal) [inline samples/endpoint/endpoint_config.yaml:1]: I think we discussed this, but can we not include all 3 of these  in a single file ? 
Something like this - [HAP Sample](https://github.com/aws-controllers-k8s/applicationautoscaling-controller/blob/main/samples/hosting-autoscaling-on-sagemaker/hosting-autoscaling-predefined.yaml)

**Review by mbaijal — APPROVED** [review]
Some minor comments which can be addressed in another iteration otherwise LGTM.
